### ISSUE TITLE
Button `onPress` does not fire when activated via Vimium

### ISSUE THREAD

**Issue by rsnay** [adobe/react-spectrum#10229]
### Provide a general summary of the issue here

Buttons built with react-aria-components Button do not call onPress when activated through [Vimium](https://github.com/philc/vimium) link hints (<kbd>f + hint key</kbd>). The button might change slightly visually, but no press handler runs.

This was reported in #6747 and closed as a duplicate of #1720 (“remove custom hit testing in usePress”). #1720 is now complete ([#7427](https://github.com/adobe/react-spectrum/pull/7427), [2025-01-15 release notes](https://react-spectrum.adobe.com/releases/2025-01-15.html)), but the Vimium case still fails in current react-aria-components.

### 🤔 Expected Behavior?

Activating a button via Vimium’s simulated click should invoke `onPress`, same as a normal mouse click or keyboard activation (Enter/Space).

This worked in a previous version after #1720, but then stopped working again recently, and does not work in the current version 1.18.0.

### 😯 Current Behavior

The button shows some visual indication of activity (but it does not become the `document.activeElement`, so is not focused), but `onPress` is never called.

### 🔦 Context

Vimium simulates clicks by dispatching a synthetic event sequence ([dom_utils.js](https://github.com/philc/vimium/blob/master/lib/dom_utils.js#L341-L355)):
```js
const eventSequence = [
  "pointerover",
  "mouseover",
  "pointerdown",
  "mousedown",
  "pointerup",
  "mouseup",
  "click",
];
```
Native <button onclick={...}> elements work with Vimium; RAC Button does not.

Related: #6747, #1720.

### 🖥️ Steps to Reproduce

1. Install Vimium [in Chrome](https://chromewebstore.google.com/detail/vimium/dbepggeogbaibhgnhhndojpepiihcmeb) or [in Firefox](https://addons.mozilla.org/en-US/firefox/addon/vimium-ff/).
2. Open https://react-aria.adobe.com/Button#pending (or any page with a RAC Button using onPress).
3. Press <kbd>f</kbd> to enter link-hint mode.
4. Press the hint key for a button (e.g., “Press me” on the Pending demo).
5. Observe a slight visual change in the button; onPress does not run.
6. Activate the button with a regular mouse click; onPress does run, and the spinner shows in the Pending demo

https://github.com/user-attachments/assets/0cbc9997-069e-4c87-98a4-b095462400b5

### Version

1.18.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

macOS 15.7.8

**Comment by snowystinger** [comment]
From a look at Vimium's source
https://github.com/philc/vimium/blob/7f4deb3f91dda66fe2aef0d4a34fc96ebef96c22/lib/dom_utils.js#L372
They need to add a few additional properties that we're expecting.

For instance, https://w3c.github.io/pointerevents/#dom-pointerevent-pressure states that there must be a pressure property on the event, and that it must be 0.5 for events that aren't pressure sensitive.
Height and Width are also required https://w3c.github.io/pointerevents/#dom-pointerevent-width

In addition, they (Vimium) probably has some other bugs. It looks like they fire off the full set of events every time. But browsers don't work like that. For instance, if you preventDefault on pointerDown, you'll cancel the compatibility mouse events.
https://jsfiddle.net/3b1nr0eh/

I don't think this is our bug. We are using what the browser spec defines.

**Comment by rsnay** [comment]
Thanks for looking into this! I appreciate the details, and will open this issue with Vimium instead.
